src/lib/x509/x509_ca.cpp

```diff
@@ -81,22 +81,25 @@ Extensions choose_extensions(const PKCS10_Request& req,
    Key_Constraints constraints;
    if(req.is_CA())
       {
-      constraints = Key_Constraints(KEY_CERT_SIGN | CRL_SIGN);
+      constraints |= Key_Constraints::KeyCertSign;
+      constraints |= Key_Constraints::CrlSign;
       }
    else
       {
-      std::unique_ptr<Public_Key> key(req.subject_public_key());
-      verify_cert_constraints_valid_for_key_type(*key, req.constraints());
       constraints = req.constraints();
       }
```

Comment:
With a few free-standing `operator|` this could be spiced up to:

```C++
const auto constraints =
  (req.is_CA())
  ? Key_Constraints::KeyCertSign | Key_Constraints::CrlSign
  : req.constraints();
```

My main objective here: Make the class "feel" like any enum type, despite the added cleverness. I.e. you'd be able to write this and still get the 'clever' type:

```C++
const auto constraints = Key_Constraints::KeyCertSign | Key_Constraints::CRL_SIGN;
BOTAN_ASSERT(!constraints.empty());
```

Godbolt: https://godbolt.org/z/oWdxKKqGf

Reply:
I went another route here, adding a `ca_constraints()` helper that returns KeyCertSign | CrlSign. That neatly solves every single case we used `|` operator outside of tests and allows use of the ternary operator as in your example.

IMO the fact that a "real" enum, `E::Val1 | E::Val2` does not result in an `E` but rather an integer, and you need to say `E(E::Val1 | E::Val2)`, it seems to me fine to require similar here.